src/types.ts

```diff
@@ -40,6 +40,19 @@ export type PackedClaims = {
 
 export type Hasher = (data: string) => Promise<string>;
 export type SaltGenerator = (size) => string;
+
+export interface IssueSDJWTOptions {
```

Comment:
@ragnika please think of a better structor for these params. Now it is a bit list and it's not easy to follow.
Example:
```
export interface IssueSDJWTOptions {
  sign: {
    alg: string;
    callback: (header, payload) => Promise<string>;
  },
  hash: {
    alg: string;
    callback: (data: string) => Promise<string>;
  }
  cnf?: JWK,
  generateSalt?: SaltGenerator;
}
```

Signer should be able to get the key itself. No point in making it an additional param to this function.
Also I believe `options` should be options and `headers` + `payload` + `disclosuers` should go as two separate params for the function.